### ISSUE TITLE
Style/ZeroLengthPredicate has false positives on Net::SFTP::Protocol::V01::Attributes objects

### ISSUE THREAD

**Issue by sandbergja** [rubocop/rubocop#12555]
The Net::SFTP gem has a class named Net::SFTP::Protocol::V01::Attributes.  It has a method named `#size` that returns an integer, but it does not have a `#empty?` method.  However, `rubocop -a` converts `attributes.size.zero?` or `attributes.size == 0` to `attributes.empty?`, which then throws an error.

--------

## Expected behavior

I would expect rubocop not to recommend using the `#empty?` method on `Net::SFTP::Protocol::V01::Attributes` or `Net::SFTP::Protocol::V04::Attributes` objects, since it doesn't exist.

## Actual behavior

```
sandbergj ~  $ rubocop --debug attributes.rb
For /Users/sandbergj: Default configuration from /Users/sandbergj/.asdf/installs/ruby/3.1.0/lib/ruby/gems/3.1.0/gems/rubocop-1.59.0/config/default.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning /Users/sandbergj/attributes.rb
C

Offenses:

attributes.rb:5:1: C: [Correctable] Style/NumericPredicate: Use attributes.size.zero? instead of attributes.size == 0.
attributes.size == 0
^^^^^^^^^^^^^^^^^^^^
attributes.rb:5:1: C: [Correctable] Style/ZeroLengthPredicate: Use empty? instead of size == 0.
attributes.size == 0
^^^^^^^^^^^^^^^^^^^^

1 file inspected, 2 offenses detected, 2 offenses autocorrectable
Finished in 0.0873539999993227 seconds
```

## Steps to reproduce the problem

1. Create a file called `attributes.rb` with this content:
    ```
    # frozen_string_literal: true
    
    require 'net/sftp'
    attributes = Net::SFTP::Protocol::V04::Attributes.new
    attributes.size == 0
    ```
1. Run `rubocop --only Style/ZeroLengthPredicate attributes.rb`
1. Expect to see no offenses

## RuboCop version

```
1.59.0 (using Parser 3.2.2.4, rubocop-ast 1.30.0, running on ruby 3.1.0) [arm64-darwin22]
```

**Comment by koic** [comment]
In RuboCop's static analysis, it is already marked as unsafe because it cannot resolve the false positives:
https://docs.rubocop.org/rubocop/1.59/cops_style.html#stylezerolengthpredicate
